packages/gcf-utils/src/gcf-utils.ts

```diff
@@ -151,6 +151,71 @@ export interface CronPayload {
   cron_org: string;
 }
 
+export interface BotSecrets {
+  privateKey: string;
+  appId: string;
+  secret: string;
+}
+
+/**
+ * A helper for fetch secret from SecretManager.
+ */
+async function getBotSecrets(): Promise<BotSecrets> {
+  const projectId = process.env.PROJECT_ID;
+  const functionName = process.env.GCF_SHORT_FUNCTION_NAME;
+  const secretsClient = new SecretManagerV1.SecretManagerServiceClient();
+  const [version] = await secretsClient.accessSecretVersion({
+    name: `projects/${projectId}/secrets/${functionName}/versions/latest`
+  });
+  // Extract the payload as a string.
+  const payload = version?.payload?.data?.toString() || '';
+  if (payload === '') {
+    throw Error('did not retrieve a payload from SecretManager.');
+  }
+  const secrets = JSON.parse(payload);
+
+  if (Object.prototype.hasOwnProperty.call(secrets, 'cert')) {
+    secrets.privateKey = secrets.cert;
+    delete secrets.cert;
+  }
+  if (Object.prototype.hasOwnProperty.call(secrets, 'id')) {
+    secrets.appId = secrets.id;
+    delete secrets.id;
+  }
+  return {
+    privateKey: secrets.privateKey,
+    appId: secrets.appId,
+    secret: secrets.secret,
+  };
+}
+
+/**
+ * A helper for getting an Octokit instance authenticated as an App.
+ */
+export async function getAuthenticatedOctokit(
+  installationId: number | null
+): Promise<Octokit> {
+  const botSecrets = await getBotSecrets();
+  if (installationId === null) {
+    return new Octokit({
+      authStragety: createAppAuth,
+      auth: {
+        appId: botSecrets.appId,
+        privateKey: botSecrets.privateKey,
+        clientSecret: botSecrets.secret,
```

Comment:
I think the stored secret is actually the webhook shared secret, not an oauth client secret

Reply:
Yeah I don't know how to authenticate it as an app. Now it's throwing an error.

Reply:
I figured out how to authenticate it as an app, but that version doesn't take care of token expiration

Reply:
I finally made it all right, I think.